core/src/epicli/data/azure/defaults/infrastructure/custom-data.yml

```diff
@@ -0,0 +1,8 @@
+kind: infrastructure/custom-data
```

Comment:
Let me explain why I've chosen such approach, then we can decide together whether it is overkill or not 😉 

1. I wanted to be able to disable cloud-init (custom-data) globally - i.e. just by changing a single setting (one line) in config file. For this reason the setting is populated from one place `infrastructure/custom-data` to all `infrastructure/virtual-machine` docs. With the alternative approach it seems it would require to change the value of `use_custom_data` for each component (VM type) and if you forget about any, you will get inconsistent environment. I run a test trying to change OS distro globally using the following docs but it was not possible - i.e. I got VM with Ubuntu.

```
kind: infrastructure/virtual-machine
name: default
provider: azure
specification:
  storage_image_reference:
    publisher: RedHat
    offer: RHEL
    sku: 7lvm-gen2
    version: latest
---
kind: infrastructure/virtual-machine
name: repository-machine-rhel
provider: azure
based_on: repository-machine
specification:
  size: Standard_DS2_v2
```
  
2. The current solution is not intuitive but quite generic (easily extendable). For example, you can add another `cloud-config` for Ubuntu, which can do something different than merging LVM. Also, it is possible to use not only `cloud-config` type but any [User-Data format](https://cloudinit.readthedocs.io/en/latest/topics/format.html) including `MIME multipart` file, which allows to specify more than one type of data. However, I don't know whether we will use cloud-init for something else than merging LVM in the future, since we prefer Ansible, so maybe this point is not meaningful.

If the current approach is not intuitive enough but we will decide to keep it, I can document how it works.

Reply:
Both fair points, so lets start with adding some documentation for sure. 

And maybe a different name for ```infrastructure/custom-data``` as for me this seems to generic?  ```infrastructure/cloud-init``` or something like that? At least that will return something usefull in google if people are looking at that and are to lazy to read the documentation:)

![image](https://user-images.githubusercontent.com/11056665/108474536-d0913880-728f-11eb-8921-55a2c5589f80.png)

Reply:
Suggestion applied, kind renamed to `infrastructure/cloud-init-custom-data` :relaxed: